Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Templates/FieldsAndConstructor.tt

```diff
@@ -8,18 +8,32 @@
     if (_model.LambdaMethod.UsingDependencyInjection)
     {
 #>
-        private readonly ServiceProvider serviceProvider;
+        private <#= _model.IsExecutable ? "static " : "" #>readonly ServiceProvider serviceProvider;
```

Comment:
The ternary operator for static should be removed since we aligned on the wrapping generated class always being an instance class.